### DISCUSSION TITLE
Offline documents.

### DISCUSSION THREAD

**Discussion by 00killer** [astubenbord/paperless-mobile#102] [Feature Requests]
Would be nice to have an offline copy of some documents like credit card, passport etc.

**Comment by schtibal** [comment]
Very useful for travel documents!  I think I've read this somewhere before, and there was a comment about security - downloading onto your device means that anyone who has access to the device could potentially gain access to your documents. But if I want to have the document offline, I'd rather have it somewhat protected, possibly with a "remove offline copy after XXX" days rather than have a PDF saved to my downloads directory or a JPG saved in my gallery

**Reply by astubenbord** [reply]
It might also be possible to store the documents encrypted, so they can only be opened in the (unlocked) app.

**Comment by kingp0dd** [comment]
Offline/cached/synced documents is very useful when you don't have internet access and need to open or search for a document. Or when you cannot connect to your server

**Comment by venci1ty** [comment]
so we are not getting Offline caching ? The other day i got into this uncomfortable situation where i needed my documents , and i was happily going to the APP to find them and.... loading screen, because no internet.  There are many reasons why we are left without internet in the wild. Needing documents that will show up only after internet connection is crazy unreliable . 

Apps like Bitwarden don't need internet to show you the passwords ,so after encryption paperless can be safe as is .. Thanks

**Reply by Caibuk** [reply]
Check out this post on ProgrammerHumor.io: The Open Source Entitlement Syndrome https://programmerhumor.io/programming-memes/the-open-source-entitlement-syndrome-7hlq

**Reply by venci1ty** [reply]
I did not mean to sound like this

**Reply by cr0n76** [reply]
I don't want to offend, but what I think is really funny at the comic is that @astubenbord transparently explained that he won't do anything in the next future. You literally would have to code it yourself 😁

See here: https://github.com/astubenbord/paperless-mobile/discussions/456#discussion-7390880

**Comment by astubenbord** [comment]
Coming back to this, I'm not sure to what extent this should be supported since it introduces a complexity regarding synchronization of local and server state which I actually wanted to avoid. That was also a big part of why I refactored a large portion of the app.

It might make sense to persist the queries in such a way that when no internet connection is present the previous state can still be displayed/recovered? But then again, that's something I actually wanted to avoid.
Or just mark individual (possibly important?) documents as "available offline" and download to a separate cache? Just some thoughts.

**Comment by kingp0dd** [comment]
The second option is a good idea if that's more feasible.
